### ISSUE TITLE
Implement faux scrolling

### ISSUE THREAD

**Issue by PaulBGD** [jwilm/alacritty#550]
Well my example uses two examples. I can use the mouse wheel to scroll inside of Vim, but scrolling inside of `htop` or `man` doesn't work. 

Let me know if you need any more information.

**Comment by 3goliad** [comment]
Alacritty does not support scrollback by design. See #124

**Comment by PaulBGD** [comment]
@3goliad Let me rephrase that, certain applications are not handling my mouse's scroll wheel.

**Comment by 3goliad** [comment]
@PaulBGD Oof, I was hoping nobody saw that. My apologies, I read your issue a little too quickly.
Could it be possible that htop and man are using an older version of ncurses?

**Comment by PaulBGD** [comment]
Well they're the latest versions of both provided for Ubuntu 16.10, even if they're a little outdated I'd think that it'd be good for alacritty to have support for them anyways.

**Comment by 3goliad** [comment]
I'm not sure whether or not this is relevant, but [this debian issue](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=838376) references an issue with scrolling using the ncurses version provided in 16.10. Which terminal emulators are displaying the correct behavior?

**Comment by NickeZ** [comment]
I can reproduce this. I can also scroll `man man` in gnome-terminal but not in alacritty.

**Comment by siiptuo** [comment]
VTE library used for example by GNOME and XFCE terminals emulates scrolling for applications without mouse support. From its [CHANGELOG](https://github.com/GNOME/vte/blob/92da93ab5aa8ee32335dfb14aacd3597823a103f/NEWS#L734):

> New feature: mouse scroll-wheel now feeds three arrow-up/down keys to the
>   terminal if in the "alternate" mode.  The alternate mode is used by apps
>   like vim, less, emacs, screen, etc.  This makes the scroll-wheel usable
> in a state that it was of no use before.

**Comment by quininer** [comment]
konsole can also scroll `man man`.

**Comment by algesten** [comment]
It seems other terminals implement a hack to make this happen. `man` uses the default pager (set by either `$MANPAGER` or `$PAGER` system env), which often defaults to `less`. 

But `less` doesn't know anything about mouse scrolling, though [I found this](https://superuser.com/questions/961719/how-to-make-less-pager-respond-to-scroll-wheel-and-not-clear-the-screen#972232):

> Many terminal emulators figured out that when it's in alternate screen mode and the application running inside is not interested in mouse events, it makes sense to convert scrolling into Up or Down keypress events.

So the question is whether @jwilm want that hack to be implemented also in alacritty?

**Comment by jwilm** [comment]
Since it seems to be standard behavior in other emulators, we should probably add it.
